sklearn/linear_model/tests/test_base.py

```diff
@@ -207,6 +216,47 @@ def test_linear_regression_sparse_multiple_outcome(random_state=0):
     assert_array_almost_equal(np.vstack((y_pred, y_pred)).T, Y_pred, decimal=3)
 
 
+def test_linear_regression_positive():
+    # Test nonnegative LinearRegression on a simple dataset.
+    # a simple dataset
+    X = [[1], [2]]
+    Y = [1, 2]
+
+    reg = LinearRegression(positive=True)
+    reg.fit(X, Y)
+
+    assert_array_almost_equal(reg.coef_, [1])
+    assert_array_almost_equal(reg.intercept_, [0])
+    assert_array_almost_equal(reg.predict(X), [1, 2])
+
+    # test it also for degenerate input
+    X = [[1]]
+    Y = [0]
+
+    reg = LinearRegression(positive=True)
+    reg.fit(X, Y)
+    assert_allclose(reg.coef_, [0])
+    assert_allclose(reg.intercept_, [0])
+    assert_allclose(reg.predict(X), [0])
+
+
+def test_linear_regression_positive_multiple_outcome(random_state=0):
+    # Test multiple-outcome nonnegative linear regressions
+    random_state = check_random_state(random_state)
+    X, y = make_sparse_uncorrelated(random_state=random_state)
+    Y = np.vstack((y, y)).T
+    n_features = X.shape[1]
+
+    ols = LinearRegression(positive=True)
+    ols.fit(X, Y)
+    assert ols.coef_.shape == (2, n_features)
+    assert np.all(ols.coef_ >= 0.)
+    Y_pred = ols.predict(X)
```

Comment:
please avoid using both Y_pred and y_pred.

Reply:
This is the same as in test [test_linear_regression_multiple_outcome](https://github.com/scikit-learn/scikit-learn/blob/ca05ce626df4ba1ebaf8336fec95f0c91e4d4af0/sklearn/linear_model/tests/test_base.py#L196) and [test_linear_regression_sparse_multiple_outcome](https://github.com/scikit-learn/scikit-learn/blob/ca05ce626df4ba1ebaf8336fec95f0c91e4d4af0/sklearn/linear_model/tests/test_base.py#L213). Should I change those ones too?

Reply:
@NicolasHug I'd like to keep the variable naming for consistency with other tests. If there is a consensus about changing those names I will be happy to propose a PR to change all the names at the same time.

Reply:
Thanks for the refs, I know remember that we use an upper case `Y` to denote a multi-output target, so `Y_pred` and `y_pred` are reasonable here.

However I see that some changes introduced here use `Y` to denote a single-output target, please use lower case instead ;)

Reply:
I'm probably missing something, but the name of the test is `test_linear_regression_positive_multiple_outcome`, so I'm assuming that it is meant to test multioutput targets. I have run it separately and `Y_pred` is a (100, 2) shape array while `y_pred` is a (100,) shape array as expected.... what I'm doing wrong?

Reply:
by "here" I mean this PR

I'm referring to e.g. `Y = [1, 2]`, `Y = [0]` etc. in other tests

Reply:
Done.